src/app/services/xml-parsers/apparatus-entries-parser.service.ts

```diff
@@ -0,0 +1,131 @@
+  }
+
+  private parseAppEntries(appEntries: XMLElement[]) {
+    this.witParserService.witnesses$.subscribe(data => this.parsedWitnesses = data);
+    const parsedEntries = appEntries.map((appEntry) => this.parseAppEntry(appEntry));
+
+    return arrayToMap(parsedEntries, 'id');
+  }
+
+  private parseAppEntry(appEntry: XMLElement): ApparatusEntry {
+    this.significantRdgNumber = 0;
+
+    return {
+      type: 'ApparatusEntryComponent',
+      id: appEntry.getAttribute('xml:id') || xpath(appEntry),
+      attributes: this.genericParserService.parseAttributes(appEntry),
+      content: this.parseAppReadings(appEntry),
+      note: this.parseAppNotes(appEntry),
+      variance: this.getVariance(),
+      originalEncoding: getOuterHTML(appEntry),
+    };
+  }
+
+  private parseAppReadings(appEntry: XMLElement): ApparatusReading[] {
+    return Array.from(appEntry.querySelectorAll(`${this.lemmaTagName}, ${this.readingTagName}`))
+      .map((child: XMLElement) => this.parseAppReading(child));
+  }
+
+  private parseAppReading(rdg: XMLElement): ApparatusReading {
+    return {
+      type: 'ApparatusReadingComponent',
+      tagname: rdg.tagName,
+      id: rdg.getAttribute('xml:id') || xpath(rdg),
+      attributes: this.genericParserService.parseAttributes(rdg),
+      wits: this.parseReadingWitnesses(rdg) || [],
+      content: this.parseAppReadingContent(rdg),
+      significant: this.parseReadingSignificance(rdg),
+    };
+  }
+
+  private parseReadingWitnesses(rdg: XMLElement) {
+    return rdg.getAttribute('wit')?.split('#')
+      .map((el) => removeSpaces(el))
+      .filter((el) => el.length !== 0);
+  }
+
+  private parseAppReadingContent(rdg: XMLElement) {
+    return Array.from(rdg.childNodes)
+      .map((child: XMLElement) => {
+        if (child.nodeName === this.appEntryTagName) {
+          return {
+            type: 'ApparatusEntryComponent',
+            id: child.getAttribute('xml:id') || xpath(child),
+            attributes: {},
+            content: [],
+          };
+        }
+
+        return this.genericParserService.parse(child);
+      });
+  }
+
+  private parseReadingSignificance(rdg: XMLElement) {
+    const notSignificantReadings = AppConfig.evtSettings.edition.notSignificantVariant;
+    let isSignificant = true;
+
+    if (notSignificantReadings.length > 0) {
+      isSignificant = this.isSignificantReading(notSignificantReadings, rdg.attributes);
+      if (isSignificant && rdg.parentElement.tagName === this.readingGroupTagName) {
+        isSignificant = this.isSignificantReading(notSignificantReadings, rdg.parentElement.attributes);
+      }
+    }
+
+    return isSignificant;
+  }
+
+  private isSignificantReading(notSignificantReading: string[], attributes: NamedNodeMap) {
+    return !Array.from(attributes).some(({name, value}) => {
+      return notSignificantReading.includes(`${name} =  ${value}`);
+    });
+  }
+
+  private parseAppNotes(appEntry: XMLElement) {
```

Comment:
Can the notes parse be used instead?

Reply:
Do you mean the parser in generic-parser? If so, no, because that parser takes an XMLElement as parameter and not an array.